source/simulator/assemblers/advection.cc

```diff
@@ -479,23 +484,43 @@ namespace aspect
 
       typename DoFHandler<dim>::face_iterator face = cell->face (face_no);
 
-      //interior face - no contribution on RHS
-      Assert (cell->neighbor(face_no).state() == IteratorState::valid,
+      //interior face or periodic face - no contribution on RHS
+
+      const typename DoFHandler<dim>::cell_iterator
+      neighbor = cell->neighbor_or_periodic_neighbor (face_no);
+      //note: "neighbor" defined above is NOT active_cell_iterator, so this includes cells that are refined
+      //for example: cell with periodic boundary.
+      Assert (neighbor.state() == IteratorState::valid,
               ExcInternalError());
-      const typename DoFHandler<dim>::cell_iterator neighbor = cell->neighbor(face_no); //note: NOT active_cell_iterator, so this includes cells that are refined.
+      const bool cell_has_periodic_neighbor = cell->has_periodic_neighbor (face_no);
 
       if (!(face->has_children()))
         {
-          if (!cell->neighbor_is_coarser(face_no) &&
+          const bool neighbor_or_periodic_neighbor_is_coarser =
+            ( cell_has_periodic_neighbor
+              ?
+              cell->periodic_neighbor_is_coarser(face_no)
+              :
+              cell->neighbor_is_coarser(face_no));
+
+          if (!neighbor_or_periodic_neighbor_is_coarser &&
+              // neighbor->level () == cell->level () &&
```

Comment:
I defined a similar bool constant "neighbor_or_periodic_neighbor_is_coarser" as the original "neighbor_is_coarser", but I just realized in the line 506, calling neighbor->level () == cell->level () does the same as calling " !neighbor_or_periodic_neighbor_is_coarser "  (are they the same?), but the former call doesn't need to define an extra bool constant "neighbor_or_periodic_neighbor_is_coarser"

So I personally prefer to use only neighbor->level () == cell->level (),  any comment?